homeassistant/components/rpi_gpio.py

```diff
@@ -5,64 +5,100 @@
 https://home-assistant.io/components/rpi_gpio/
 """
 import logging
+import importlib
+
+import voluptuous as vol
 
 from homeassistant.const import (
     EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['RPi.GPIO==0.6.5']
+REQUIREMENTS = ['RPi.GPIO==0.6.5',
+                'OPi.GPIO==0.3.6']
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_BOARD_FAMILY = 'board_family'
+
+DEFAULT_FAMILY = 'raspberry_pi'
+FAMILY_LIBRARIES = {'raspberry_pi': 'RPi.GPIO',
+                    'orange_pi': 'OPi.GPIO'}
+
 DOMAIN = 'rpi_gpio'
+LIBRARY = 'gpio_library'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_BOARD_FAMILY, default=DEFAULT_FAMILY): cv.string
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
 
+class UnknownBoardFamily(Exception):
+    """board_family should be 'raspberry_pi' or 'orange_pi'."""
 
-def setup(hass, config):
-    """Set up the Raspberry PI GPIO component."""
-    from RPi import GPIO  # pylint: disable=import-error
+    pass
+
+
+def setup(hass, base_config):
+    """Set up the GPIO component."""
+    hass.data[DOMAIN] = {}
+
+    config = base_config.get(DOMAIN)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.